File: event-service/src/main/java/com/github/manosbatsis/services/event/rest/dto/UserEventResponse.java
package com.github.manosbatsis.services.event.rest.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonGetter;

import java.util.Date;

public record UserEventResponse(
        Long userId,
        @JsonFormat(
                        shape = JsonFormat.Shape.STRING,
                        pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSZ",
                        timezone = "UTC")
                Date datetime,
        String type,
        String data) {

    @JsonGetter("id")
    public String getId() {
        return userId + "-" + datetime.getTime();
    }
}

File: event-service/src/main/kotlin/com/github/manosbatsis/services/event/rest/dto/UserEventResponse.kt
package com.github.manosbatsis.services.event.rest.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.util.Date

data class UserEventResponse(
    val userId: Long,
    @JsonFormat(
        shape = JsonFormat.Shape.STRING,
        pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSZ",
        timezone = "UTC",
    )
    val datetime: Date,
    val type: String,
    val data: String,
) {
    val id: String = userId.toString() + "-" + datetime.time
}
